src/appmixer/kafkadavid/jobs.js

```diff
@@ -0,0 +1,97 @@
+
+            connections.debug(context, {
+                type: 'syncConnectionsJob',
+                status: 'syncing',
+                registeredConnections,
+                openConnections
+            });
+
+            await Promise.allSettled(registeredConnections.forEach(async (conn) => {
+                const connectionId = conn.key;
+                const connectionParameters = conn.value;
+                if (!openConnections[connectionId]) {
+                    // The connection is not created on this node in the cluster. Create it but before that check
+                    // if it's really needed (still registered in the cluster).
+                    const stillNeeded = await context.service.stateGet(connectionId);
+
+                    connections.debug(context, {
+                        type: 'syncConnectionsJob',
+                        status: 'notopen-creating',
+                        conn
+                    });
+
+                    if (stillNeeded) {
+                        await context.log('info', `Connecting component: ${connectionId}`);
+                        if (connectionId.startsWith('consumer:')) {
+                            await connections.addConsumer(
+                                context,
+                                connectionParameters.topics,
+                                connectionParameters.flowId,
+                                connectionParameters.componentId,
+                                connectionParameters.groupId,
+                                connectionParameters.fromBeginning,
+                                connectionParameters.auth
+                            );
+                        } else {
+                            await connections.addProducer(
+                                context,
+                                connectionParameters.flowId,
+                                connectionParameters.componentId,
+                                connectionParameters.auth
+                            );
+                        }
+                    }
+                }
+            }));
+
+            // If the connection is live on this node but it is not desired anymore (not registered in the cluster), remove it.
+            await Promise.allSettled(Object.keys(openConnections).forEach(async (connectionId) => {
+                const conn = await context.service.stateGet(connectionId);
+                if (!conn) {
+                    if (connectionId.startsWith('consumer')) {
+                        await connections.removeConsumer(context, conn.flowId, conn.componentId);
+                    } else {
+                        await connections.removeProducer(context, conn.flowId, conn.componentId);
+                    }
+                }
+            }));
+
+        } catch (error) {
+            await context.log('error', `Error occurred during connection sync job: ${error.message}`);
```

Comment:
<img width="1382" alt="appmixer-1-gmlmw_·_Pod_·_Logs_·_OKD" src="https://github.com/clientIO/appmixer-connectors/assets/928772/08b76107-fbf9-47e8-bc04-fa1b1cb13e6d">

I saw this error in the logs of `learning-gull-30931` tenant.